Microsoft.Toolkit.Uwp/Helpers/BluetoothLEHelper/BluetoothLEHelper.cs

```diff
@@ -0,0 +1,350 @@
+                "System.Devices.Aep.ContainerId",
+                "System.Devices.Aep.DeviceAddress",
+                "System.Devices.Aep.IsConnected",
+                "System.Devices.Aep.IsPaired",
+                "System.Devices.Aep.IsPresent",
+                "System.Devices.Aep.ProtocolId",
+                "System.Devices.Aep.Bluetooth.Le.IsConnectable"
+            };
+
+            // BT_Code: Currently Bluetooth APIs don't provide a selector to get ALL devices that are both paired and non-paired.
+            _deviceWatcher = DeviceInformation.CreateWatcher(
+                BluetoothLeDeviceWatcherAqs,
+                requestedProperties,
+                DeviceInformationKind.AssociationEndpoint);
+
+            // Register event handlers before starting the watcher.
+            _deviceWatcher.Added += DeviceWatcher_Added;
+            _deviceWatcher.Updated += DeviceWatcher_Updated;
+            _deviceWatcher.Removed += DeviceWatcher_Removed;
+            _deviceWatcher.EnumerationCompleted += _deviceWatcher_EnumerationCompleted;
+
+            _advertisementWatcher = new BluetoothLEAdvertisementWatcher();
+            _advertisementWatcher.Received += AdvertisementWatcher_Received;
+
+            BluetoothLeDevices.Clear();
+
+            _deviceWatcher.Start();
+            _advertisementWatcher.Start();
+        }
+
+        /// <summary>
+        /// Stops enumeration of bluetooth device
+        /// </summary>
+        public void StopEnumeration()
+        {
+            if (_deviceWatcher != null)
+            {
+                _deviceWatcher.Added -= DeviceWatcher_Added;
+                _deviceWatcher.Updated -= DeviceWatcher_Updated;
+                _deviceWatcher.Removed -= DeviceWatcher_Removed;
+
+                if (EnumerationCompleted != null)
+                {
+                    _deviceWatcher.EnumerationCompleted -= _deviceWatcher_EnumerationCompleted;
+                }
+
+                _deviceWatcher.Stop();
+                _deviceWatcher = null;
+            }
+
+            if (_advertisementWatcher != null)
+            {
+                _advertisementWatcher.Received -= AdvertisementWatcher_Received;
+                _advertisementWatcher.Stop();
+                _advertisementWatcher = null;
+            }
+        }
+
+        /// <summary>
+        /// An event for when the enumeration is completed.
+        /// </summary>
+        /// <param name="sender">The sender.</param>
+        /// <param name="args">The args.</param>
+        private void _deviceWatcher_EnumerationCompleted(DeviceWatcher sender, object args)
+        {
+            EnumerationCompleted?.Invoke(sender, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Initializes the app context.
+        /// </summary>
+        private async void Init()
+        {
+            _adapter = await BluetoothAdapter.GetDefaultAsync();
+            Context = new BluetoothLEHelper();
```

Comment:
Seems weird creating another BluetoothLEHelper here when it's already created by this point